onnx/defs/tensor/defs.cc

```diff
@@ -1979,6 +1992,11 @@ ONNX_OPERATOR_SET_SCHEMA(
                 } else if (indices_shape.dim(i).has_dim_param()) {
                   dim->set_dim_param(indices_shape.dim(i).dim_param());
                 }
+              } else if (i == axis) {
+                if (depth_name != "")
+                  dim->set_dim_param(depth_name);
+                else
+                  dim->set_dim_value(depth);
```

Comment:
This looks off in two ways.

If we were able to read an actual 'depth' value shouldn't that be preferred?

If we didn't read a value for 'depth' from an initializer, I would have expected we shouldn't call set_dim_value to the default of 0. I'm assuming 'depth' doesn't necessarily come from an initializer (so the getInputData call may return nullptr) so the actual value may not be known until runtime.

Reply:
Agree with Scott. Furthermore, we should not reuse a tensor's name as a dimension-variable (they belong to different namespaces and this could introduce an unintended clash if the model used the same name for another dimension-variable). We should leave the dim with neither param nor value set if we don't know the depth value.